api/src/main/java/jakarta/persistence/criteria/ComparableExpression.java

```diff
@@ -0,0 +1,138 @@
+     * @return greater-than-or-equal predicate
+     * @see CriteriaBuilder#greaterThanOrEqualTo(Expression, Expression)
+     */
+    Predicate greaterThanOrEqualTo(Expression<? extends Comparable<? super X>> y);
+
+    /**
+     * Create a predicate for testing whether this expression is
+     * greater than or equal to the first argument.
+     * @param y  value
+     * @return greater-than-or-equal predicate
+     * @see CriteriaBuilder#greaterThanOrEqualTo(Expression, Comparable)
+     */
+    <Y extends Comparable<? super X>> Predicate greaterThanOrEqualTo(Y y);
+
+    /**
+     * Create a predicate for testing whether this expression is
+     * less than the first argument.
+     * @param y  expression
+     * @return less-than predicate
+     * @see CriteriaBuilder#lessThan(Expression, Expression)
+     */
+    Predicate lessThan(Expression<? extends Comparable<? super X>> y);
+
+    /**
+     * Create a predicate for testing whether this expression is
+     * less than the first argument.
+     * @param y  value
+     * @return less-than predicate
+     * @see CriteriaBuilder#lessThan(Expression, Comparable)
+     */
+    <Y extends Comparable<? super X>> Predicate lessThan(Y y);
+
+    /**
+     * Create a predicate for testing whether this expression is
+     * less than or equal to the first argument.
+     * @param y  expression
+     * @return less-than-or-equal predicate
+     * @see CriteriaBuilder#lessThanOrEqualTo(Expression, Expression)
+     */
+    Predicate lessThanOrEqualTo(Expression<? extends Comparable<? super X>> y);
+
+    /**
+     * Create a predicate for testing whether this expression is
+     * less than or equal to the first argument.
+     * @param y  value
+     * @return less-than-or-equal predicate
+     * @see CriteriaBuilder#lessThanOrEqualTo(Expression, Comparable)
+     */
+    <Y extends Comparable<? super X>> Predicate lessThanOrEqualTo(Y y);
+
+    /**
+     * Create a predicate for testing whether this expression is
+     * between the first and second arguments in value.
+     * @param x  expression
+     * @param y  expression
+     * @return between predicate
+     * @see CriteriaBuilder#between(Expression, Expression, Expression)
+     */
+    Predicate between(Expression<? extends Comparable<? super X>> x, Expression<? extends Comparable<? super X>> y);
+
+    /**
+     * Create a predicate for testing whether this expression is
+     * between the first and second arguments in value.
+     * @param x  value
+     * @param y  value
+     * @return between predicate
+     * @see CriteriaBuilder#between(Expression, Comparable, Comparable)
+     */
+    <Y extends Comparable<? super X>> Predicate between(Y x, Y y);
+
+    //aggregate functions:
+
+    /**
+     * Create an aggregate expression for finding the greatest of
+     * the values (strings, dates, etc).
+     * @return greatest expression
+     * @see CriteriaBuilder#greatest(Expression)
+     */
+    ComparableExpression<X> max();
```

Comment:
Add `greatest` and `least` as well (non-aggegrate functions)? Or aren't these considered standard enough?